physics/cires_ugwp_solv2_v1_mod.F90

```diff
@@ -0,0 +1,810 @@
+module cires_ugwp_solv2_v1_mod
+
+
+contains
+
+
+!---------------------------------------------------
+!  Broad spectrum FVS-1993, mkz^nSlope with nSlope = 0, 1,2 
+!  dissipative solver with NonHyd/ROT-effects
+!  reflected GWs treated as waves with "negligible" flux,
+!  they are out of given column
+!---------------------------------------------------
+      subroutine cires_ugwp_solv2_v1(im, levs, dtp    ,          &
+                 tm , um, vm, qm, prsl, prsi, zmet,  zmeti,  &
+                 prslk, xlatd, sinlat, coslat,                   &
+                 pdudt, pdvdt, pdtdt, dked, tauabs, wrms, trms,  &
+                 tau_ngw, mpi_id, master, kdt)
+!
+!--------------------------------------------------------------------------------
+!      nov 2015 alternative gw-solver for nggps-wam
+!      nov 2017 nh/rotational gw-modes for nh-fv3gfs
+!      oct 2019 adding empirical satellite-based
+!          source function and *F90 CIRES-style of the code
+! --------------------------------------------------------------------------------
+!  
+ 
+      use machine,          only : kind_phys  
+
+      use cires_ugwp_module_v1,only :  krad, kvg, kion, ktg 
+      
+      use cires_ugwp_module_v1,only :  knob_ugwp_doheat, knob_ugwp_dokdis, idebug_gwrms
+      
+      use ugwp_common_v1 ,     only : rgrav,  grav,  cpd,    rd,  rv, rcpdl, grav2cpd,    &
```

Comment:
Constants should come from the host model through the argument list, if possible.

Reply:
Done.